sammba/registration/struct.py

```diff
@@ -582,22 +581,38 @@ def anats_to_common(anat_filenames, write_dir, brain_volume,
                     iwarp=True,
                     weight=nonlinear_weight_file,
                     iniwarp=[warp_file],
-                    inilev=levels_minpatches[n_iter - 1] + 1,
-                    maxlev=level_or_minpatch,
+                    inilev=nonlinear_levels[n_iter - 1] + 1,
+                    maxlev=maxlev,
                     out_file=out_file,
                     **verbosity_quietness_kwargs)
-                    
-            else:
-                out_qwarp = qwarp2(
-                    in_file=centered_head_file,
-                    base_file=nwarp_adjusted_mean,
-                    noneg=True,
-                    iwarp=True,
-                    weight=nonlinear_weight_file,
-                    iniwarp=[warp_file],
-                    inilev=nonlinear_levels[-1] + 1, # not ideal
-                    minpatch=level_or_minpatch,
-                    out_file=out_file)
+
+           warped_files.append(out_qwarp.outputs.warped_source)
+           warp_files.append(out_qwarp.outputs.source_warp)
+           previous_warp_files = warp_files
+  
+    for n_iter, minpatch in enumerate(nonlinear_minimal_patches):
+        
+        if n_iter == 0:
+            previous_warp_files = affine_transform_files
```

Comment:
shouldn't this be previous_warp_files = warp_files or even better previous_warp_files = warp_files.copy() ? (this bug type where a pointer is created instead of data copied may be present elsewhere in struct.py). otherwise the non-linear warp will be initialised to the affine transform rather than the latest non-linear transform calculated in the previous loop.

Reply:
yes you are right. It should be the last warp_files produced from the loop before, unless `nonlinear_levels` is an empty list.
@marinacelestine I suggest you remove the `n_iter` and the condition `if n_iter == 0`, and you replace it with testing `if nonlinear_levels == []`, in which case you should have `previous_warp_files = affine_transform_files` and `inilev=0`. Otherwise, `inilev` would be `nonlinear_levels[-1] + 1` and `previous_warp_files` do not need to be defined